src/cli/cli.cc

```diff
@@ -1627,31 +1724,25 @@ int main (const int argc, const char* argv[]) {
     exit(0);
   });
 
-  createSubcommand("list-devices", { "--platform", "--ecid", "--udid", "--only" }, false, [&](const std::span<const char *>& options) -> void {
-    String targetPlatform = "";
-    bool isUdid = false;
-    bool isEcid = false;
-    bool isOnly = false;
-    for (auto const& option : options) {
-      auto platform = optionValue("list-devices", options[0], "--platform");
-      if (platform.size() > 0) {
-        targetPlatform = platform;
-      }
-      if (is(option, "--udid") || is(rc["list-devices_udid"], "true")) {
-        isUdid = true;
-      }
-      if (is(option, "--ecid") || is(rc["list-devices_ecid"], "true")) {
-        isEcid = true;
-      }
-      if (is(option, "--only") || is(rc["list-devices_only"], "true")) {
-        isOnly = true;
-      }
-    }
-
-    if (targetPlatform.size() == 0) {
-      log("ERROR: --platform option is required");
-      exit(1);
-    }
+  // flag indicating whether option is optional
+  std::map<std::vector<String>, bool> listDevicesOptions = {
+    { { "--platform" }, false }, // non-optional
+    { { "--ecid" }, true },
+    { { "--udid" }, true },
+    { { "--only" }, true }
+  };
+  createSubcommand("list-devices", listDevicesOptions, false, [&](Map argumentsWithValue, std::unordered_set<String> argumentsWithoutValue) -> void {
+    bool isUdid =
+      argumentsWithValue.find("--udid") != argumentsWithValue.end() ||
```

Comment:
Perhaps what we can implement in a follow up PR is to define a list of arguments and how they are used.
There would then be one or two arg retrieval functions that encapsulate the logic of finding an argument.
Ideally, the `.find()` and `!= .end()` type calls should be hidden from someone who just needs to add a new option.

Reply:
I agree